handlers/authentication.go

```diff
@@ -699,14 +713,29 @@ func OpenIdCodeFlowHandler(w http.ResponseWriter, r *http.Request) bool {
 		return true
 	}
 
-	// Check if user trying to login has enough privileges to login
-	httpStatus, errMsg, detailedError := business.VerifyOpenIdUserAccess(openIdParams.IdToken)
-	if detailedError != nil {
-		RespondWithDetailedError(w, httpStatus, errMsg, detailedError.Error())
-		return true
-	} else if httpStatus != http.StatusOK {
-		RespondWithError(w, httpStatus, errMsg)
-		return true
+	if conf.Auth.OpenId.DisableRBAC {
+		// When RBAC is on, we delegate some validations to the Kubernetes cluster. However, if RBAC is off
+		// the token must be fully validated, as we no longer pass the OpenId token to the cluster API server.
+		// Since the configuration indicates RBAC is off, we do the validations:
+		err = business.ValidateOpenTokenInHouse(openIdParams)
+		if err != nil {
+			RespondWithDetailedError(w, http.StatusForbidden, "the OpenID token was rejected", err.Error())
+			return true
+		}
+	}
+
+	if !conf.Auth.OpenId.DisableRBAC {
```

Comment:
You could just use a else here, or short circuit on the first if, keeping this one without a conditional.

Reply:
My eyes didn't saw that... I'll change...